compiler/rustc_trait_selection/src/solve/inspect.rs

```diff
@@ -183,18 +224,24 @@ impl<'tcx> ProofTreeBuilder<'tcx> {
         is_normalizes_to_hack: IsNormalizesToHack,
     ) -> ProofTreeBuilder<'tcx> {
         if self.state.is_none() {
-            return ProofTreeBuilder { state: None };
+            return ProofTreeBuilder {
+                state: None,
+                disable_global_cache: self.disable_global_cache,
+            };
         }
 
-        ProofTreeBuilder::new(WipGoalEvaluation {
-            uncanonicalized_goal: goal,
-            canonicalized_goal: None,
-            evaluation_steps: vec![],
-            is_normalizes_to_hack,
-            cache_hit: None,
-            returned_goals: vec![],
-            result: None,
-        })
+        ProofTreeBuilder::new(
```

Comment:
vibe: `self.nested(WipGoalEvaluation { .. })`?

don't have to propagate `diable_global_cache` with that